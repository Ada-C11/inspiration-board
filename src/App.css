
* {
  box-sizing: border-box;
}

body {
  font-family: 'Raleway', sans-serif;
  background: rgba(0, 213, 85, 0.938);
}

h1, h2 {
  font-family: 'Permanent Marker', Helvetica, sans-serif;
}

.header {
  padding: 4rem;
}

.header__h1 {
  margin: auto 0;

  text-align: center;
}

.header__text {
  padding: 1rem;
  display: inline-block;
  background-color:white;
  color: rgb(0, 0, 0);
  text-transform: uppercase;
  font-size: 3rem;
  border-style: inset;

  transform: rotate(-3deg) skew(-3deg);
}
