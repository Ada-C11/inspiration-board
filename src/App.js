import React, { Component } from 'react';
import './App.css';
import Board from './components/Board';

class App extends Component {

  constructor(){
    super();

    this.state = {
      errorMessage: null
    };
  }

  showErrorMessage = (error) => {
    console.log('API response error: ', error);
    this.setState({ errorMessage: error.message });
    // const errorList = this.state.errorMessage;
    //   const newError = error.response.data.errors.text;
    //   newError.forEach((text) => {
    //     errorList.push(text);
    //   })
    //   this.setState({ errorMessage: errorList });
  }


  render() {
    const errorSection = (this.state.errorMessage) ?
    (<section className="error">
       Error: {this.state.errorMessage}
     </section>) : null;
    return (
      <section>
        <header className="header">

        {errorSection}

          <h1 className="header__h1"><span className="header__text"> Elle's Inspirations</span></h1>
        </header>
        <Board
          url="https://inspiration-board.herokuapp.com/boards/"
          boardName={`elle`}
          showErrorMessageCallback={this.showErrorMessage}
          />
      </section>
    );
  }
}

export default App;

// {
//     "board": {
//         "id": 63,
//         "name": "elle"
//     }
// }

// API Documentation:
// https://github.com/AdaGold/inspiration-board-api
