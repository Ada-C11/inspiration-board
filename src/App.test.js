import React from 'react';
// import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';


describe('<App />', () => {  
  test('that it renders App with shallow rendering', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

    // Remove the component from the DOM (save memory and prevent side effects).
    // wrapper.unmount();
});
