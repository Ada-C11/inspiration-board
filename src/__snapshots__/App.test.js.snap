// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> that it renders App with shallow rendering 1`] = `
<section>
  <header
    className="header"
  >
    <h1
      className="header__h1"
    >
      <span
        className="header__text"
      >
        Inspiration Board
      </span>
    </h1>
  </header>
  <Board
    boardName="Faiza"
    url="https://inspiration-board.herokuapp.com/"
  />
</section>
`;

exports[`<App /> will match the last snapshot with deep rendering 1`] = `
<App>
  <section>
    <header
      className="header"
    >
      <h1
        className="header__h1"
      >
        <span
          className="header__text"
        >
          Inspiration Board
        </span>
      </h1>
    </header>
    <Board
      boardName="Faiza"
      url="https://inspiration-board.herokuapp.com/"
    >
      <div
        className="board"
      >
        <NewCardForm
          addCardCallback={[Function]}
        >
          <form
            className="new-card-form"
            onSubmit={[Function]}
          >
            <h3>
               Add a Card
            </h3>
            <div>
              <label
                htmlFor="text"
              >
                 Text: 
              </label>
              <input
                name="text"
                onChange={[Function]}
                type="text"
                value=""
              />
              <label
                htmlFor="emoji"
              />
              <select
                name="emoji"
                onChange={[Function]}
                type="text"
                value=""
              >
                <option
                  value=""
                >
                  Add emojis to your message!
                </option>
                <option
                  value=""
                >
                  None
                </option>
                <option
                  value="heart_eyes"
                >
                  Heart Eyes
                </option>
                <option
                  value="beer"
                >
                  Beer
                </option>
                <option
                  value="clap"
                >
                  Clap
                </option>
                <option
                  value="sparkling_heart"
                >
                  Sparkling Heart
                </option>
                <option
                  value="heart_eyes_cat"
                >
                  Heart Eyes Cat
                </option>
                <option
                  value="dog"
                >
                  Dog
                </option>
              </select>
            </div>
            <div
              className=".new-card-form__form-button"
            >
              <input
                className="btn btn-success new-card-form--submit"
                name="submit"
                type="submit"
                value="Add a Card"
              />
            </div>
          </form>
        </NewCardForm>
      </div>
    </Board>
  </section>
</App>
`;
